relay_test.go

```diff
@@ -1420,6 +1420,33 @@ func TestRelayTransferredBytes(t *testing.T) {
 	})
 }
 
+func TestRelayCallResponse(t *testing.T) {
```

Comment:
Can we add an integration test that sends response headers, and validates them via the `RespFrameFn`?

Reply:
not sure if i'm missing something - in this test the response header is set by the echo handler, copied over from the call, and we verify that the bytes in arg2 are exactly what we send. is there something else you had in mind?

Reply:
What I was thinking was that instead of verifying the arg2 bytes, we actually response with thift response headers so arg2 is an encoded header map, and we validate the values map.

I think we do something like that on the request side:
https://github.com/uber/tchannel-go/blob/dev/relay_test.go#L1233